src/pip/_internal/req/req_install.py

```diff
@@ -180,6 +183,9 @@ def __init__(
         # This requirement needs more preparation before it can be built
         self.needs_more_preparation = False
 
+        # This requirement needs to be unpacked before it can be installed.
+        self.archive_source: Optional[Path] = None
```

Comment:
@uranusjr how do you feel about adding a mutable `archive_source` attribute to `InstallRequirement`? It allows us to avoid changing anywhere else in `prepare.py`.